pkg/operator/controllers/subnets/subnet_serviceendpoint.go

```diff
@@ -12,10 +12,19 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 )
 
 func (r *reconcileManager) ensureSubnetServiceEndpoints(ctx context.Context, s subnet.Subnet) error {
+	// Do not add/reconcile service endpoints if egress lockdown is enabled.
+	if gatewayEnabled(r.instance) {
+		r.log.Info("Skipping service endpoint reconciliation since egress lockdown is enabled")
```

Comment:
Same feedback here - since (most?) clusters are egress lockdown enabled, we might want to reverse this and log on that so we don't add too many new operator logs.